### ISSUE TITLE
Stack on errors does not map to the original source

### ISSUE THREAD

**Issue by Glinkis** [oven-sh/bun#7120]
### What version of Bun is running?

1.0.11+f7f6233ea

### What platform is your computer?

Darwin 23.0.0 x86_64 i386

### What steps can reproduce the bug?

`test.ts`
```ts
const e = new Error();
console.log(e.stack);
throw e;
```

```sh
bun run test.ts
```

### What is the expected behavior?

The stack should point at the same original line in the stack as when the error is thrown.
```
Error: 
    at module code (/Users/<user>/Documents/Private/<project>/test.ts:2:19)
1 | const e = new Error();
2 | console.log(e.stack);
                      ^
error: Error
      at /Users/<user>/Documents/Private/<project>/test.ts:2:19
```

### What do you see instead?

```
Error: 
    at module code (/Users/vicgli/Documents/Private/<project>/test.ts:1:19)
1 | const e = new Error();
2 | console.log(e.stack);
                      ^
error: Error
      at /Users/vicgli/Documents/Private/<project>/test.ts:2:19
```

### Additional information

If I log out the error instead of only the stack, I get the correct line.

```
error: Error
      at /Users/<user>/Documents/Private/<project>/test.ts:1:10

1 | const e = new Error;
             ^
error: Error
      at /Users/<user>/Documents/Private/<project>/test.ts:1:10
```

**Comment by Electroid** [comment]
Related: #6373

**Comment by ssalbdivad** [comment]
Unfortunately this breaks Bun's source maps in [attest](https://github.com/arktypeio/arktype/tree/beta/ark/attest#readme).

Copying some context here from https://github.com/arktypeio/arktype/issues/880:

> Currently, attest type assertions cannot be executed from Bun because the stack they attach to an Error is incorrect.
> 
> You can see the correct trace by `console.log`ging it, but I'm not sure how to get access to that output at runtime.
> 
> If anyone has any suggestions, please let me know, otherwise we'll have to wait for [oven-sh/bun#7120](https://github.com/oven-sh/bun/issues/7120) to be resolved.

**Comment by Jarred-Sumner** [comment]
Fixed in Bun v1.1.20. 
```ts
/tmp
❯ bunx bun@1.0.11 test.ts
Error:
    at module code (/private/tmp/test.ts:1:19)
1 | const e = new Error();
2 | console.log(e.stack);
                      ^
error: Error
      at /private/tmp/test.ts:2:19

/tmp
❯ bunx bun@1.1.0 test.ts
Error
    at module code (/private/tmp/test.ts:1:19)
    at moduleEvaluation (:1:11)
    at moduleEvaluation (:1:11)
    at <anonymous> (:2:1)
1 | const e = new Error();
                      ^
error: Error
      at module code (/private/tmp/test.ts:1:19)

/tmp
❯ bunx bun@1.1.10 test.ts
Error
    at module code (/private/tmp/test.ts:1:8)
    at moduleEvaluation (:1:11)
    at moduleEvaluation (:1:11)
    at <anonymous> (:2:1)
1 | const e = new Error();
           ^
error: Error
      at module code (/private/tmp/test.ts:1:8)

Bun v1.1.10 (macOS arm64)

/tmp took 2s
❯ bunx bun@1.1.20 test.ts
Error
    at module code (/private/tmp/test.ts:1:11)
    at moduleEvaluation (:1:11)
    at moduleEvaluation (:1:11)
    at <anonymous> (:2:1)
1 | const e = new Error();
              ^
error: Error
      at module code (/private/tmp/test.ts:1:11)

Bun v1.1.20 (macOS arm64)
```

If you find other places where this occurs, please leave a comment with some code to reproduce it.
